### PR TITLE
Entire docset review

### PR DESCRIPTION
**Pull request by tconsigny**
This will be the running PR for the docset review. Tasks:

- Review all docstrings
- Ask questions about any unclear parameters, etc.
- Investigate and hopefully implement Python decorators for docstrings shared across functions

### CODE REVIEW

File: verticapy/_utils/_map.py
Comment (tconsigny) [inline verticapy/_utils/_map.py:77]: @oualib
This is one option for using decorators to reuse parameter descriptions. In this case, you'd create a dictionary of parameter descriptions for each file (see classification.py) and then supply that dictionary and the parameters of the function to the decorator. The decorator would then insert the parameter descriptions between the function description and the listed return value.  
So, you would only need to define each parameter once in the parameter dictionary. Then all you need to do is supply the necessary parameter keys to the decorator.
To use the decorator, you'd just need to import the function add_docstring (didn't where this should go, so put it in _utils._map.py for now)
I don't know how this impacts function runtime, or how it interacts with nested decorators. 
This is my best shot for now. If this is a bad approach, just let me know and I'll review the doc as normal and come back to this sometime in the future.

File: verticapy/_utils/_map.py
Comment (tconsigny) [inline verticapy/_utils/_map.py:122]: @oualib 
This is one option for using decorators to reuse parameter descriptions. In this case, you'd create a dictionary of parameter descriptions for each file (see classification.py) and then supply that dictionary and the parameters of the function to the decorator. The decorator would then insert the parameter descriptions between the function description and the listed return value.  
So, you would only need to define each parameter once in the parameter dictionary. Then all you need to do is supply the necessary parameter keys to the decorator.
To use the decorator, you'd just need to import the function add_docstring (didn't where this should go, so put it in _utils._map.py for now)
I don't know how this impacts function runtime, or how it interacts with nested decorators. 
This is my best shot for now. If this is a bad approach, just let me know and I'll review the doc as normal and come back to this sometime in the future.

File: verticapy/machine_learning/metrics/classification.py
Comment (tconsigny) [inline verticapy/machine_learning/metrics/classification.py:77]: @oualib 
Example decorator and parameter dictionary

File: verticapy/_utils/_map.py
Comment (oualib) [reply verticapy/_utils/_map.py]: Nice, can you please check the render when you use the 'help' function on one of the function? Is it well aligned?

Comment (tconsigny) [reply verticapy/_utils/_map.py]: Looks like it renders alright:


help-printout

Comment (oualib) [reply verticapy/_utils/_map.py]: Nice!

**Comment by oualib** [comment]
For the test which is failing, do not investigate some of the tests are not stable and they sometimes return different results. Your decorator is super cool!

**Comment by tconsigny** [comment]
Update. I'm onto the final section (machine_learning). Should finish up either later today, or sometime tomorrow.

**Comment by tconsigny** [comment]
So I've gone through all the functions. There were only a couple of questions I had, which I'll comment on below. 
If the decorator looks good, I'll create a follow-up issue for converting the docstrings over. Didn't want to
hold up the process now, so thought it best to tackle this task separately. 
And I just want to confirm that any new notebooks will be reviewed in the VerticaPy-Website PR as usual?

File: verticapy/core/vdataframe/_multiprocessing.py
Comment (tconsigny) [inline verticapy/core/vdataframe/_multiprocessing.py:27]: I couldn't tell whether these were intended to be internal or not. If they are going to be public, might need some help w/ the descriptions for these two functions

File: verticapy/plotting/base.py
Comment (tconsigny) [inline verticapy/plotting/base.py:234]: Same as above comment

File: verticapy/plotting/_utils.py
Comment (tconsigny) [inline verticapy/plotting/_utils.py:71]: Same as above

**Comment by oualib** [comment]
hi @tconsigny is it done or (work in progress) if yes then erase the mention from the PR.

**Comment by tconsigny** [comment]
> hi @tconsigny is it done or (work in progress) if yes then erase the mention from the PR.

Ah yeah forgot to remove that from the PR title. It's done. Once this merges, I'll create a follow-up issue for the doc decorator.

**Comment by tconsigny** [comment]
> > hi @tconsigny is it done or (work in progress) if yes then erase the mention from the PR.
> 
> Ah yeah forgot to remove that from the PR title. It's done. Once this merges, I'll create a follow-up issue for the doc decorator.

bump

**Review by oualib — APPROVED** [review]
We still need to complete the missing doc in another PR.

**Comment by oualib** [comment]
@tconsigny can you please do another PR to indicate where some doc string are missing? So I can comment and we can finalise the doc?

I'll merge this PR.

**Pull request merged**
